exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java

```diff
@@ -102,4 +112,34 @@ public void dispose(DrillUserPrincipal principal) {
       // No-Op
     }
   }
+
+  // Provider which creates and cleanups DrillUserPrincipal for anonymous (auth disabled) mode
+  public static class AnonDrillUserPrincipalProvider implements Factory<DrillUserPrincipal> {
+    @Inject WorkManager workManager;
+
+    @RequestScoped
+    @Override
+    public DrillUserPrincipal provide() {
+      return new AnonDrillUserPrincipal(workManager.getContext());
+    }
+
+    @Override
+    public void dispose(DrillUserPrincipal principal) {
+      // If this worked it would have been clean to free the resources here, but there are various scenarios
```

Comment:
Original issue: https://java.net/jira/browse/JERSEY-2299. It got fixed in 2.7 (we are on 2.8), but I saw few threads that mention it still reproes and not always called when closing the browser (http://jersey.576304.n2.nabble.com/Factory-dispose-is-not-called-td7583326.html). Didn't find any latest JIRAs. When I tried also, it never called after the request is complete. So thought it is not safe to rely on this method to cleanup resources.